landing/src/dojo/modelManager/ConfigManager.ts

```
@@ -441,10 +470,62 @@ export class ClientConfigManager {
+  getHyperstructureRequiredAmounts(hyperstructureId: number) {
+    const hyperstructure = getComponentValue(
+      this.components.Hyperstructure,
+      getEntityIdFromKeys([BigInt(hyperstructureId)]),
+    );
+
+    const randomness = BigInt(hyperstructure?.randomness ?? 0);
+    const requiredAmounts: { resource: ResourcesIds; amount: number }[] = [];
+
+    // Get amounts for each tier
+    for (const tier in ResourceTier) {
+      if (isNaN(Number(tier))) continue; // Skip non-numeric enum values
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)